### PR TITLE
Updates to install and configure sections

### PR DESCRIPTION
**Pull request by jasonblais**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,40 +23,62 @@ Originally developed by [Romain Maneschi](https://github.com/manland).
...
-5. (Optional) Lock the plugin to a GitLab group
-    * Go to System Console -> Plugins -> GitLab and set the GitLab Group field to the name of your GitLab group
-6. (Optional) Enable private repositories
-    * Go to System Console -> Plugins -> GitLab and set Enable Private Repositories to true
-    * Note that if you do this after users have already connected their accounts to GitLab they will need to disconnect and reconnect their accounts to be able to use private repositories
-7. Enable the plugin
-    * Go to System Console -> Plugins -> Management and click "Enable" underneath the GitLab plugin
-8. Test it out
-    * In Mattermost, run the slash command `/gitlab connect`
+In Mattermost 5.14 and later, the GitLab plugin is pre-packaged and no steps are required for installation. You can go directly to [Configuration](#configuration).
```
Comment (aaronrothschild) [inline README.md]:
```suggestion
In Mattermost 5.16 and later, the GitLab plugin is included in the Plugin Marketplace which can be accessed from **Main Menu > Plugins Marketplace**. You can install the GitLab plugin and then configure it via the [Plugin Marketplace "Configure" button](#configuration).
```

File: README.md [hunk 2]
```diff
@@ -23,40 +23,62 @@ Originally developed by [Romain Maneschi](https://github.com/manland).
...
+   - Go to System Console -> Plugins -> GitLab and set the GitLab Group field to the name of your GitLab group
+3. (Optional) Enable private repositories
+   - Go to System Console -> Plugins -> GitLab and set Enable Private Repositories to true
+   - Note that if you do this after users have already connected their accounts to GitLab they will need to disconnect and reconnect their accounts to be able to use private repositories
+4. Enable the plugin
+   - Go to System Console -> Plugins -> Management and click "Enable" underneath the GitLab plugin
+5. Test it out
+   - In Mattermost, run the slash command `/gitlab connect`
+
+1. Go to **System Console > Plugins > GitLab** and do the following:
```
Comment (aaronrothschild) [inline README.md]:
```suggestion
1. Go to **Plugin Marketplace > GitLab > Configure** and do the following:
```

File: README.md [hunk 3]
```diff
@@ -23,40 +23,62 @@ Originally developed by [Romain Maneschi](https://github.com/manland).
...
+
+### Step 1: Register an OAuth application in GitLab
+   
+1. Go to https://gitlab.com/profile/applications or https://gitlab.yourdomain.com/profile/applications to register an OAuth app.
+2. Set the following values:
+   - **Name**: `Mattermost GitLab Plugin - <your company name>`
+   - **Redirect URI**: `https://your-mattermost-url.com/plugins/com.github.manland.mattermost-plugin-gitlab/oauth/complete`, replacing `https://your-mattermost-url.com` with your Mattermost URL
+3. Select `api` and `read_user` in **Scopes**
+4. Save the application. Copy the **Application ID* and **Secret** fields in the resulting screen.
+2. In Mattermost, go to **System Console > Plugins > GitLab**, and enter the **GitLab URL**, **GitLab OAuth Client ID**, and **Gitlab OAuth Client Secret**
```
Comment (aaronrothschild) [inline README.md]:
```suggestion
2. In Mattermost, go to **Plugins Marketplace > GitLab > Configure**, and enter the **GitLab URL**, **GitLab OAuth Client ID**, and **Gitlab OAuth Client Secret**
```

File: README.md [hunk 4]
```diff
@@ -23,40 +23,62 @@ Originally developed by [Romain Maneschi](https://github.com/manland).
...
+
+### Step 2: Create a GitLab webhook
+
+__Note for each project you want to receive notifications for or subscribe to, you must create a webhook__
+
+1. In Mattermost, go to **System Console > Plugins > GitLab**, generate a new value for **Webhook Secret**. Copy it as you will use it in a later step.
```
Comment (aaronrothschild) [inline README.md]:
```suggestion
1. In Mattermost, go to **Plugins Marketplace > GitLab > Configure**, generate a new value for **Webhook Secret**. Copy it as you will use it in a later step.
```

File: README.md [hunk 5]
```diff
@@ -23,40 +23,62 @@ Originally developed by [Romain Maneschi](https://github.com/manland).
...
+5. Test it out
+   - In Mattermost, run the slash command `/gitlab connect`
+
+1. Go to **System Console > Plugins > GitLab** and do the following:
+  - Generate a new value for **At Rest Encryption Key**.
+  - (Optional) **GitLab Group**: Lock the plugin to a single GitLab group by setting this field to the name of your GitLab group.
+  - (Optional) **Enable Private Repositories**: Allow the plugin to receive notifications from private repositories by setting this value to true.
+    When enabled, existing users must reconnect their accounts to gain access to private project. Affected users will be notified by the plugin once private repositories are enabled.
+2. Hit **Save**.
+3. Go to **System Console > Plugins > Management** and click **Enable** to enable the GitLab plugin.
```
Comment (aaronrothschild) [inline README.md]:
```suggestion
3. Go to **Plugins Marketplace > GitLab > Configure > Enable Plugin** and click **Enable** to enable the GitLab plugin.
```

**Comment by jasonblais** [comment]
@aaronrothschild updates made

**Comment by jasonblais** [comment]
@manland Any chance you've had a chance to review? 🙏

**Comment by manland** [comment]
> @manland Any chance you've had a chance to review? pray

@jasonblais I'm sorry, I have do it when you have made this PR https://community.mattermost.com/core/pl/iu581hd7j3giigctosfxmujsac but i'm never use github, and I miss the `submit review`... :no_good_man: I didn't understand why you don't react to my comments... sorry!

**Comment by jasonblais** [comment]
Thanks @manland! I made some changes accordingly: https://github.com/mattermost/mattermost-plugin-gitlab/pull/115/commits/fcce5a7612c4c610786e04c13b0d59b9370ddb82

**Comment by manland** [comment]
thanks @jasonblais ;)

**Pull request merged**